### ISSUE TITLE
agent preferred name - policy request

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#8872]
## can we adopt a policy/documentation/Best Practices/something of choosing an informative preferred name when possible

---------------

I'm trying to un-mess a mess and ran across a pair of string-duplicates:

[H. Smith](https://arctos.database.museum/agent/10011105) is a no-data string probably dredged out of the darkest corners of MVZ Archives.

[H. Smith](https://arctos.database.museum/agent/21357931) is a recent creation with a bunch of very good data, and what looks to me like a poorly chosen primary name destined to be missed and misused.

For "available" agents, I think our policy should be that their preferred name is whatever they prefer, end of discussion.

When there's not a handy person to ask, can we adopt a policy/documentation/Best Practices/something of choosing an informative preferred name when possible?

@camwebb help?

**Comment by dustymc** [comment]
See also [N. Hotchkiss](https://arctos.database.museum/agent.cfm?agent_name==N.%20Hotchkiss)

**Comment by camwebb** [comment]
Fixed the Preferred names for both these. I'll alert Elena about this issue.

**Comment by dustymc** [comment]
@mkoo @ArctosDB/agents-committee (exists??) I noticed https://arctos.database.museum/agent/21368127 (via error logs) - can we add this to documentation (and @ridabrams is the cryptic name necessary?)

**Comment by Jegelewicz** [comment]
https://handbook.arctosdb.org/best_practices/Agents.html also exists and it includes

> Do format person agent preferred names as “First Middle Last” as a matter of standardization.
> Do “assemble” namestrings when possible. “A. Lincoln” is an acceptable (if not very good) preferred name, but also include AKA name “Abraham Lincoln.” The existence of name components (eg, “first name”) is not sufficient to prevent the creation of duplicates.

Which seem like we aren't being clear. Can we not just say that preferred name should be the most complete name possible? Would simply adding the aka "Jan Willem Hoetmer" to that agent be a sufficient solution? If so, could we also REQUIRE that either the preferred name match a concatenation of first middle last names OR some aka should match? OR can Arcto just create an aka that is the concatenation?

There are probably TONS of agents just like the example. It would be nice if the solution could be automated.

**Comment by dustymc** [comment]
>aren't being clear

So how do we fix that?

>aka "Jan Willem Hoetmer" to that agent be a sufficient solution?

I don't think so, people will create duplicates unless things are hard to avoid (which probably suggests something about operator training that we can't address here...).

>REQUIRE

Pretty hard no from me, if https://arctos.database.museum/agent/21368127 insists on being called {whatever} then we should accommodate, but we should also encourage/soft-require up-front/preferred name completeness when we can.

>Arcto just create

No, names aren't that simple. (But possibly Arctos could suggest/make it easy to add more.)

**Comment by camwebb** [comment]
> Can we not just say that preferred name should be the most complete name possible?

Since the preferred name is what appears on labels, etc., my preference has been to use 'First M. I. Last', unless there is an identical preferred name for another agent.  Most readers of labels don't want all the full names.

**Comment by dustymc** [comment]
>labels

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctagent_attribute_type#labels

**Comment by Jegelewicz** [comment]
In addition to what @dustymc pointed out about labels, I think the other issue is what people are seeing in Arctos. I believe preferred name is what shows up anywhere an agent appears on an Arctos page. Are people annoyed at seeing Jan Willem Hoetmer instead of J. W. Hoetmer? I know we have discussed this in the past, but should we also include some way for preferred name to explicitly be the most complete but offer a display name for people who don't want to have the whole thing show up everywhere?  Honestly, I am always baffeled at an agent that has some weird initials in preferred name, but nice complete names in first, middle and last names. Why disguise a person in that way?

**Comment by DerekSikes** [comment]
Are people annoyed at seeing Jan Willem Hoetmer instead of J. W. Hoetmer? 

I'm not annoyed & in Arctos, on the screen, I can't see any problems with unabbreviated names.

I've broken with entomological tradition to using full names on our labels when we have the space (more commonly this is true for our vial labels). Avoiding future ambiguity is a good thing! I see two advantages to abbreviated names: space savings for labels & matching to common strings for data entry folks. If all the old labels say C. W. Harrington and a data entry tech searches on Harrington and finds Charles Washington Harrington (even if that agent has an aka C. W. Harrington) they might be unsure that is the C. W. Harrington they need & perhaps want to make a new one to avoid matching to the wrong person. We can't expect every data entry tech to have the confidence of whomever did the research to find the full names of that person.

This reminds me of something I've said before - and a solution to the above 2nd issue: for specimens with labels made outside of Arctos it would be great if we had a way to clearly indicate "this is the agent text string on the label" and "this is the person the data entry person thinks that name matches" side-by-side in Arctos in the catalog record _in all the places_ (ID agent, collector, georeferencer, etc). So when looking at the agent name we'd see both strings adjacent and clearly indicated. 

BUT... as easy as this is to say, I'm sure it has countless issues with increased complexity, slower data entry, harder searches, etc. Maybe not worth it?  The experiment with verbatim agents left me a bit shaken.

**Comment by dustymc** [comment]
>clearly indicate "this is the agent text string on the label" and "this is the person the data entry person thinks that name matches" side-by-side in Arctos

That's a core use case of https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#verbatim_agent - the "experiment" was only in **not** creating the accompanying low-data agents when the "label name" is already carrying all of the available information.

**Comment by dustymc** [comment]
>preferred name is what shows up anywhere an agent appears on an Arctos page

Yep.

>offer a display name 

Display name **is** preferred name, there can be no functional/useful distinction (for Arctos-wide usage, of course anyone with a report or API key can do whatever they want however they want to do it...)

>don't want to have the whole thing show up everywhere

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctagent_attribute_type#aka (or possibly https://arctos.database.museum/info/ctDocumentation.cfm?table=ctagent_attribute_type#birth_name) is suitable for that.

I'm asking if we want to push that towards more exposure when we can, because I suspect people don't bother looking beyond preferred name, because the data keeps leading me to that idea. I don't think we need anything beyond policy. "Just say that preferred name should be the most complete name possible" (with some exceptions/guidance for 'please call me...' &etc. use cases) is about right, I think

Here's a view I was playing with, in case it's useful for something: https://docs.google.com/spreadsheets/d/1DmsdhzNf5e5FYO8KX7AmCuMzou4LRtS2lt1utd_2D88/edit?gid=806747360#gid=806747360

Most-ish agents seem to have some sort of nearly-full-ish name variant, but that still leaves many thousands of paths to duplicates and chaos - I didn't have to scroll far to find https://arctos.database.museum/agent.cfm?srch=Walter%20Wheaton%20Dimmick&include_verbatim=false&include_bad_dup=false, which fails to find https://arctos.database.museum/agent/21267648.

There are also tons of https://arctos.database.museum/agent/21311917 - I'm not sure if ``R.`` is useful as a first or middle name or not, or if asking the creator to make that leap is defensible, but that should probably be discussed for any developing policy (and maybe bots, see below).

Also lots of https://arctos.database.museum/agent/679 - I'd suggest "Bud" should also be a first name, and that "Bud M...." might be generated - but two first names would probably also break any potential bots, so ?? - possibly something to plan and discuss.

I very much dislike https://arctos.database.museum/agent/5346, and there's now a non-breaking place for uncertainty (attribute method/remarks, which didn't exist when that was entered).

>can Arcto just create

I think I did change my view of this somewhere along the way. Names are now attributes, attributes are assertions, assertions have determiners, that's a very different situation than when Sybil (proto-Arctos) was generating (confusing) names as "facts." I think we could set up a bot to do whatever, and it would be easy enough to ignore, delete, or rebuild any messes created by such an Agent.

**Comment by dustymc** [comment]
Above comments seem overall supportive of this, and as @Jegelewicz  points out it was mostly already in https://handbook.arctosdb.org/best_practices/Agents.html. I made some very mild edits to that, and flagged some of the examples. (And maybe we need some sort of 'uses flagged agents' report, someone file an Issue if so.)

https://github.com/ArctosDB/arctos/issues/9409 is the "bots might..." thread.

https://github.com/ArctosDB/arctos/issues/9410 is the "embedded uncertainty" thread.

Closing.

**Comment by camwebb** [comment]
@dustymc How is the `labels` attribute of an Agent exposed in `flat`?  If we use `flat.collectors` it shows the preferred name. It seems that it would take some complicated SQL to check every collector agent for a `labels` field, and rebuild the collectors string.  Also, unless I've missed it, `flat.collectors` does not include verbatim agents, which is a barrier to getting people to using verbatim agents.

A solution to this would be to generate a second collectors field in `flat` which would i) use the agents `labels` field if present, and ii) append `verbatim_agent`s at the end.  If there is support for this I can open a new issue.

**Comment by dustymc** [comment]
>flat

It's not, and I'd suggest stashing it there wouldn't be a great use of (very limited) resources - there aren't many, they can be used in a LOT of ways, they're probably not very stable, and I don't anticipate any issues with pulling them on the fly for reports.

Let me know what you want and I can set up the SQL....

>flat.collectors does not include verbatim agents

No, they're not collectors, but they're in attributedetail - and I can pull those in whatever form you find useful too, just let me know some details.
